packages/components/addon/components/hds/copy/button/index.js

```diff
@@ -0,0 +1,108 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Component from '@glimmer/component';
+import { assert } from '@ember/debug';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+
+export const DEFAULT_SIZE = 'medium';
+export const SIZES = ['small', 'medium'];
+export const DEFAULT_ICON = 'clipboard-copy';
+export const SUCCESS_ICON = 'clipboard-checked';
+
+export default class HdsCopyButtonComponent extends Component {
+  @tracked isSuccess = false;
+  @tracked isError = false;
+  @tracked icon = DEFAULT_ICON;
+
+  /**
+   * @param clipboardText
+   * @type {string}
+   * @description The ID of the element containing the text to be copied. If no clipboardText value is defined `isError` will be set to true.
+   */
+  get clipboardText() {
+    let { clipboardText } = this.args;
+    return clipboardText;
+  }
+
+  /**
+   * @param size
+   * @type {string}
+   * @default medium
+   * @description The size of the copy/button; acceptable values are `small` and `medium`
+   */
+  get size() {
+    let { size = DEFAULT_SIZE } = this.args;
+
+    assert(
+      `@size for "Hds::Copy::Button" must be one of the following: ${SIZES.join(
+        ', '
+      )}; received: ${size}`,
+      SIZES.includes(size)
+    );
+
+    return size;
+  }
+
+  /**
+   * Get the class names to apply to the component.
+   * @method CopyButton#classNames
+   * @return {string} The "class" attribute to apply to the component.
+   */
+  get classNames() {
+    let classes = ['hds-copy-button'];
+
+    // add a class based on the @size argument
+    classes.push(`hds-copy-button--size-${this.size}`);
+
+    // add a class based on the @color argument
+    classes.push(`hds-copy-button--color-secondary`);
+
+    return classes.join(' ');
+  }
+
+  @action
+  async copyCode() {
+    let textToCopy;
+    if (this.args.clipboardText) {
+      let clipboardTextContent = document
+        .querySelector(this.args.clipboardText)
```

Comment:
if the consumer provides a wrong selector, this will fail with a rather cryptic message
<img width="717" alt="screenshot_2917" src="https://github.com/hashicorp/design-system/assets/686239/1c52ca80-0ce0-4461-b6b7-28f3fe6a95e3">

we should show an error (to the consumer, not the end user) that the targeted element doesn't exist